support/src/deposit.rs

```diff
@@ -29,18 +32,29 @@ pub struct Deposit<Account, Balance> {
 	pub amount: Balance,
 }
 
-pub fn reserve_deposit<Account, Currency: ReservableCurrency<Account>>(
+pub(crate) fn reserve_deposit<Account, Currency: Mutate<Account>>(
 	account: Account,
 	deposit_amount: Currency::Balance,
+	reason: &Currency::Reason,
 ) -> Result<Deposit<Account, Currency::Balance>, DispatchError> {
-	Currency::reserve(&account, deposit_amount)?;
+	Currency::hold(reason, &account, deposit_amount)?;
 	Ok(Deposit {
 		owner: account,
 		amount: deposit_amount,
 	})
 }
 
-pub fn free_deposit<Account, Currency: ReservableCurrency<Account>>(deposit: &Deposit<Account, Currency::Balance>) {
-	let err_amount = Currency::unreserve(&deposit.owner, deposit.amount);
-	debug_assert!(err_amount.is_zero());
+pub(crate) fn free_deposit<Account, Currency: Mutate<Account>>(
+	deposit: &Deposit<Account, Currency::Balance>,
+	reason: &Currency::Reason,
+) -> DispatchResult {
+	let result = Currency::release(reason, &deposit.owner, deposit.amount, Precision::BestEffort);
+	debug_assert!(
+		result == Ok(deposit.amount),
+		"Released deposit amount does not match with expected amount. Expected: {:?}, Released amount: {:?}  Error: {:?}",
+		deposit.amount,
+		result.unwrap(),
```

Comment:
This will still fail if `result` is not `Ok`.

Reply:
Is it not intended to fail if `result` is `Err`?
Anyway, I think the function is redundant and changed it in: [68aa397](https://github.com/KILTprotocol/kilt-node/pull/527/commits/68aa397dd20c60435c6128e2a18497c62c08e1c1)